dev/ProgressRing/ProgressRing.cpp

```diff
@@ -84,29 +121,70 @@ void ProgressRing::OnBackgroundPropertyChanged(const winrt::DependencyObject&, c
 
 void ProgressRing::OnBackgroundColorPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
 {
-    if (auto&& player = m_player.get())
+    if (auto&& determinatePlayer = m_determinatePlayer.get())
+    {
+        if (auto const progressRingDeterminate = determinatePlayer.Source())
+        {
+            SetLottieBackgroundColor(progressRingDeterminate);
+        }
+    }
+
+    if (auto&& indeterminatePlayer = m_indeterminatePlayer.get())
     {
-        if (auto const progressRingIndeterminate = player.Source().try_as<AnimatedVisuals::ProgressRingIndeterminate>())
+        if (auto const progressRingIndeterminate = indeterminatePlayer.Source())
         {
             SetLottieBackgroundColor(progressRingIndeterminate);
         }
     }
 }
 
+void ProgressRing::OnOpacityPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
+{
+    if (auto&& indeterminatePlayer = m_indeterminatePlayer.get())
+    {
+        auto test = indeterminatePlayer.Opacity();
+
+        if (indeterminatePlayer.Opacity() == 0)
+        {
+            indeterminatePlayer.Stop();
+        }
+    }
+}
+
 void ProgressRing::OnIsActivePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
 {
-    ChangeVisualState();
+    UpdateStates();
+}
+
+void ProgressRing::OnIsIndeterminatePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateStates();
 }
 
 void ProgressRing::SetAnimatedVisualPlayerSource()
 {
-    if (auto&& player = m_player.get())
+    if (auto&& determinatePlayer = m_determinatePlayer.get())
+    {
+        if (!determinatePlayer.Source())
+        {
+            determinatePlayer.Source(winrt::make<AnimatedVisuals::ProgressRingDeterminate>());
```

Comment:
>winrt::make<AnimatedVisuals::ProgressRingDeterminate>() [](start = 37, length = 55)

nit: slightly better to make this as a local and set the properties you want on it so that you don't have to call the players .Source() getter